keanu-project/src/main/java/io/improbable/keanu/vertices/ConstantVertexFactory.java

```diff
@@ -10,9 +10,9 @@
 import io.improbable.keanu.vertices.generic.nonprobabilistic.ConstantGenericVertex;
 import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
 
-public class ConstantVertex {
+public class ConstantVertexFactory {
 
-    private ConstantVertex() {
+    private ConstantVertexFactory() {
```

Comment:
This is a factory, fair, but do we have to put that in its name? It's very Enterprise Java (J2EE) to go for MyClassFactoryFactory as a name. Is there a reason or is it just to highlight the specific pattern?

Reply:
as a matter of fact you didn't need to rename `ConstantVertex` at all. Just make it an interface - you're allowed static methods on an interface.

Reply:
Done